src/Typedefs/TermParse.idr

```diff
@@ -107,28 +113,128 @@ injection (FS FZ) [a, b]             x = Right x
 injection FZ     (a :: b :: c :: ts) x = Left x
 injection (FS i) (a :: b :: c :: ts) x = Right (injection i (b :: c :: ts) x)
 
-deserializeBinary : (t : TDefR n) -> (ts : Vect n (a ** Deserialiser a)) -> Deserialiser (Ty (map DPair.fst ts) t)
-deserializeBinary T0 ts = fail -- will never happen!
-deserializeBinary T1 ts = pure ()
-deserializeBinary td@(TSum {k = k} tds) ts = do
-  i <- deserializeInt (2 + k)
-  t' <- deserializeBinary (assert_smaller td (index i tds)) ts
+deserialiseBinary : (t : TDefR n) -> (ts : Vect n (a ** Deserialiser a)) -> Deserialiser (Ty (map DPair.fst ts) t)
+deserialiseBinary T0 ts = fail -- will never happen!
+deserialiseBinary T1 ts = pure ()
+deserialiseBinary td@(TSum {k = k} tds) ts = do
+  i <- deserialiseInt (2 + k)
+  t' <- deserialiseBinary (assert_smaller td (index i tds)) ts
   pure (injection i tds t')
-deserializeBinary (TProd [a, b]) ts = do
-  ta <- deserializeBinary a ts
-  tb <- deserializeBinary b ts
+deserialiseBinary (TProd [a, b]) ts = do
+  ta <- deserialiseBinary a ts
+  tb <- deserialiseBinary b ts
   pure (ta, tb)
-deserializeBinary td@(TProd (a ::  b :: c :: tds)) ts = do
-  ta <- deserializeBinary a ts
-  t' <- deserializeBinary (assert_smaller td (TProd (b :: c :: tds))) ts
+deserialiseBinary td@(TProd (a ::  b :: c :: tds)) ts = do
+  ta <- deserialiseBinary a ts
+  t' <- deserialiseBinary (assert_smaller td (TProd (b :: c :: tds))) ts
   pure (ta, t')
-deserializeBinary (TMu tds) ts = do
-  t <- assert_total $ deserializeBinary (args tds) ((Mu (map DPair.fst ts) (args tds) ** assert_total $ deserializeBinary (TMu tds) ts)::ts)
+deserialiseBinary (TMu tds) ts = do
+  t <- assert_total $ deserialiseBinary (args tds) ((Mu (map DPair.fst ts) (args tds) ** assert_total $ deserialiseBinary (TMu tds) ts)::ts)
   pure (Inn t)
-deserializeBinary (TVar FZ) (t::ts) = snd t
-deserializeBinary {n = S (S n')} (TVar (FS i)) (t::ts) = deserializeBinary {n = S n'} (TVar i) ts
-deserializeBinary (TApp (TName name def) xs) ts = assert_total $ deserializeBinary (ap def xs) (ts)
-deserializeBinary (RRef n) ts = fail -- we dont' support reference types
+deserialiseBinary (TVar FZ) (t::ts) = snd t
+deserialiseBinary {n = S (S n')} (TVar (FS i)) (t::ts) = deserialiseBinary {n = S n'} (TVar i) ts
+deserialiseBinary (TApp (TName name def) xs) ts = assert_total $ deserialiseBinary (ap def xs) (ts)
+deserialiseBinary (RRef n) ts = fail -- we dont' support reference types
+
+deserialiseBinaryClosed : (t : TDefR 0) -> Bytes-> Maybe ((Ty [] t), Bytes)
+deserialiseBinaryClosed t = runDeserialiser (deserialiseBinary t [])
+
+
+JSONM : Type -> Type
+JSONM = Either String
+
+expectObject : JSON -> JSONM (List (String, JSON))
```

Comment:
should we call it `parseObject` to align with the ones below?

Reply:
fixed